### PR TITLE
MkDocs projects use RTD's analytics privacy improvements

### PR DESCRIPTION
**Pull request by davidfischer**
Read the Docs has implemented some privacy enhancements for authors who opt to enable Google Analytics on their projects. This change enables those for MkDocs projects.

For example, the mkdocs project (a documentation project that uses mkdocs itself) specifies their own [Google Analytics code in their `mkdocs.yml`](https://github.com/mkdocs/mkdocs/blob/5ee6600/mkdocs.yml#L36). When MkDocs builds the docs, the Google analytics snippet is injected directly into the output. However, Read the Docs already supports user GA analytics codes and has enabled some improvements around there such as anonymizing visitor IPs. Without this change, MkDocs projects bypass these protections which can result in reduced privacy or possibly double counting in analytics (if the analytics code is specified in RTD and in `mkdocs.yml`). By hooking into the RTD setup, we can enable those privacy enhancements for people who specify their Google Analytics setup in `mkdocs.yml` and ensure there's no double counting.

### CODE REVIEW

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 1]
```diff
@@ -100,18 +100,23 @@ def append_conf(self, **__):
...
         docs_path = os.path.join(self.root_path, docs_dir)
 
         # RTD javascript writing
         rtd_data = self.generate_rtd_data(docs_dir=docs_dir, mkdocs_config=user_config)
         with open(os.path.join(docs_path, 'readthedocs-data.js'), 'w') as f:
             f.write(rtd_data)
 
+        # Use Read the Docs' analytics setup rather than mkdocs'
+        # This supports using RTD's privacy improvements around analytics
+        user_config['google_analytics'] = None
```
[the remark this answers is not in the document]
Comment (davidfischer) [reply readthedocs/doc_builder/backends/mkdocs.py]: I use `del` extremely rarely, but I don't have a particularly strong opinion.

File: readthedocs/doc_builder/backends/mkdocs.py [hunk 2]
```diff
@@ -120,6 +125,12 @@ def generate_rtd_data(self, docs_dir, mkdocs_config):
         if theme_dir:
             theme_name = theme_dir.rstrip('/').split('/')[-1]
 
+        # Use the analytics code from mkdocs.yml if it isn't set already by Read the Docs
+        analytics_code = self.version.project.analytics_code
+        if not analytics_code and mkdocs_config.get('google_analytics'):
+            # http://www.mkdocs.org/user-guide/configuration/#google_analytics
+            analytics_code = mkdocs_config['google_analytics'][0]
```
[the remark this answers is not in the document]
Comment (davidfischer) [reply readthedocs/doc_builder/backends/mkdocs.py]: The docs link leads me to believe it is a list. I'm fine being a bit more defensive here.

[the remark this answers is not in the document]
Comment (davidfischer) [reply readthedocs/doc_builder/backends/mkdocs.py]: It is guaranteed to be either `null` or a 2-tuple

[the remark this answers is not in the document]
Comment (ericholscher) [reply readthedocs/doc_builder/backends/mkdocs.py]: :+1:

**Pull request merged**